### PR TITLE
Jcrocker 20150515 shutdown

### PR DESCRIPTION
**Pull request by james-crocker**

### CODE REVIEW

File: src/lib/perl/OVF/State.pm [hunk 1]
```diff
@@ -547,30 +546,68 @@ sub isOvfEnvChanged ( \@ ) {
...
+	
+	# If no vmtoolUpdated; we assume that the ovf environment properties are those set from vCenter. If
+	# vmtoolUpdated exists; we assume that the vCenter environment properties have been overridden with
+	# ui changes to the ovf-defaults. So, if no vmtoolUpdated and previousVcenterPath differs from
+	# current; then vCenter had changes - override all previous. If no vmtoolUpdated and previousVcenterPath
+	# same as current, then see if ovf-defaults in play.
+
+	# No vmtoolUpdated
+	if ( !-e $vmtoolUpdated and -e $previousVcenterPath ) {
+		tie @previousOvfEnvProperties, 'Tie::File', $previousVcenterPath, autochomp => 1 or ( Sys::Syslog::syslog( 'err', qq{$action Couldn't open $previousVcenterPath ($?:$!)} ) and die );
```
Comment (clemep-sios) [inline src/lib/perl/OVF/State.pm]: I'm guessing that order of this file is consistent (keys always stored in same order), or the below code would not work (taking md5 of the contents) ?

File: src/lib/perl/OVF/State.pm [hunk 2]
```diff
@@ -547,30 +546,68 @@ sub isOvfEnvChanged ( \@ ) {
...
+				updateVmtool($previousPath);
+			}
+		}
+	} else {
+		Sys::Syslog::syslog( 'info', qq{$action NO previous vCenter OVF environment properties file [ $previousVcenterPath ] OR [ $vmtoolUpdated ] exists} );
+	}
+	
+	# If no changes from vCenter; any changes from other sources
+	if ( -e $previousPath ) { 
 		tie @previousOvfEnvProperties, 'Tie::File', $previousPath, autochomp => 1 or ( Sys::Syslog::syslog( 'err', qq{$action Couldn't open $previousPath ($?:$!)} ) and die );
```
Comment (clemep-sios) [inline src/lib/perl/OVF/State.pm]: You could avoid duplicating all this code by putting it in a loop or by making it a function.

**Comment by srazin** [comment]
please address Paul's comments. Otherwise make sense.

File: src/lib/perl/OVF/State.pm (hunk 1 above)
Comment (james-crocker) [inline src/lib/perl/OVF/State.pm]: Order is imposed by printOvfProperties for consistent md5 checking.

**Comment by james-crocker** [comment]
Ping! Updated.

**Pull request merged**